arangod/GeneralServer/CommTask.cpp

```diff
@@ -660,8 +660,6 @@ CommTask::Flow CommTask::canAccessPath(auth::TokenCache::Entry const& token,
         // `/_api/users/<name>` to check their passwords
         result = Flow::Continue;
         vc->forceReadOnly();
-      } else if (StringUtils::isPrefix(path, ApiUser)) {
-        result = Flow::Continue;
```

Comment:
I actually do not get why this is not equivalent ?

Reply:
Probably what is missing there is the extra check for `vc->databaseAuthLevel() != auth::Level::NONE`

Reply:
I debugged it and the difference is that the broken code sets the status from `Abort` to `Continue` via the now-deleted code:
```
       } else if (StringUtils::isPrefix(path, ApiUser)) {
         result = Flow::Continue;
       }
```
The correct behavior (as in 3.6 and the code of this PR) is to not make this change, and leave the status on `Abort` as decided here:
``` 
Flow result = userAuthenticated ? Flow::Continue : Flow::Abort;

// deny access to database with NONE
if (vc->databaseAuthLevel() == auth::Level::NONE && !StringUtils::isPrefix(path, ApiUser)) {
  result = Flow::Abort;
}
```
For the particular problem, the initial status is already `Abort`, and the correct code does not change it to anything else from there.
The broken code determined the initial status as `Abort`, but then overwrote it with `Continue` in the final `else if` that this PR deletes.

Reply:
oh ok, so the problem is the user is not authenticated at all, why not just check for that later then ?
`else if (userAuthenticated && StringUtils::isPrefix(path, ApiUser))```

Reply:
Ok, did.